File: sso.js
const crypto = require('crypto-js');

/**
 * dependencies
 *  - crypto-js
 */



function hyvorTalkSignon(user) {
    const HYVOR_TALK_SSO_PRIVATE_KEY = 'MY_PRIVATE_KEY'; // this is not recommended. Use .env instead

    const userData = Buffer.from(JSON.stringify(user)).toString('base64');
    const hash = crypto.HmacSHA1(userData, HYVOR_TALK_SSO_PRIVATE_KEY).toString();

    return {
        userData,
        hash
    }
}

const signon = hyvorTalkSignon({
    id: 1,
    name: 'Example User',
    email: 'some@example.com',
    picture: 'https://example.com/picture.png',
    url: 'https://example.com'
});

const installationCode = `
<div id="hyvor-talk-view"></div>
<script type="text/javascript">
    var HYVOR_TALK_WEBSITE = YOUR_WEBSITE_ID
    var HYVOR_TALK_CONFIG = {
        url: false,
        id: false,
        sso: {
            hash: "${signon.hash}",
            userData: "${signon.userData}",
            loginURL: "https://example.com/login"
        }
    };
</script>
<script async type="text/javascript" src="//talk.hyvor.com/web-api/embed"></script>`;

// now send the installationCode to the front-end

File: sso.java
import java.util.HashMap;
import java.util.Base64;

import org.apache.commons.codec.digest.HmacAlgorithms;
import org.apache.commons.codec.digest.HmacUtils;
import org.json.simple.JSONObject;

public class Sso {

	public static void main(String[] args) {
		HashMap<String, String> signon = hyvorTalkSignon();
		
		String userData = signon.get("userData"));
		String hash = signon.get("hash"));
	}

	public static HashMap<String, String> hyvorTalkSignon() {
		String SECRET_KEY = "MY_PRIVATE_KEY";

		User user = new User();
		JSONObject userJson = new JSONObject();

		if (user.isLoggedIn) {
			userJson.put("id", user.id);
			userJson.put("name", user.name);
			userJson.put("email", user.email);
			userJson.put("picture", user.picture);
			userJson.put("url", user.url);
		}

		String userBase64 = Base64.getEncoder().encodeToString(userJson.toString().getBytes());		
		String hash = new HmacUtils(HmacAlgorithms.HMAC_SHA_1, SECRET_KEY).hmacHex(userBase64);

		HashMap<String, String> signon = new HashMap<String, String>();
		signon.put("userData", userBase64); 
		signon.put("hash", hash); 
		return signon;
	}

}

/**
* User Class example
* 
    public class User {	
        boolean isLoggedIn = true;
        
        int id = 103004;
        String name = "Example User";
        String email = "some@example.com";
        String picture = "https://example.com/picture.png";
        String url = "https://example.com";
        
    }
*/